unix: appease warning about non-standard `inline`

chisophugis opened the pull request:
Clang warns about using `inline`, which is not technically allowed in
C89 mode (libuv compiles with `-std=c89`). It's probably best to leave
it to the compiler to do the inlining anyway.

The exact warnings that Clang emits are:

```
../../deps/uv/src/unix/fs.c:500:8: warning: extension used [-Wlanguage-extension-token]
static inline void uv__to_stat(struct stat* src, uv_stat_t* dst) {
       ^
1 warning generated.
```

```
../../deps/uv/src/unix/signal.c:119:1: warning: extension used [-Wlanguage-extension-token]
inline static uv_signal_t* uv__signal_first_handle(int signum) {
^
1 warning generated.
```

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

The following commiters were not found in the CLA:
- Sean Silva

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

chisophugis committed:
unix: appease warning about non-standard `inline`

Clang warns about using `inline`, which is not technically allowed in
C89 mode (libuv compiles with `-std=c89`). It's probably best to leave
it to the compiler to do the inlining anyway.

chisophugis commented:
Derp, used my other email address, which is not the one I signed the CLA with (I contributed this tiny patch to node a while back https://github.com/joyent/node/pull/5332). Should be good now.

bnoordhuis commented:
Thanks, landed in c396486.
